amads/melody/mel_accent.py

```diff
@@ -0,0 +1,129 @@
+__author__ = "Yiwen Zhao"
+
+import numpy as np
+from ..core.basics import Score, Note, Part
+
+def calculate_melodic_accent(score: Score) -> list:
+    """
+    Calculate melodic accent salience according to Thomassen's model.
+    
+    Melodic accent is determined by analyzing patterns of three consecutive notes.
+    The model assigns different accent values based on the following patterns:
+    - Peak accent (up-down pattern): 1.0
+    - Valley accent (down-up pattern): 0.6
+    - Continuous ascending: 0.4
+    - Continuous descending: 0.4
+    - Other patterns (including repeated notes): 0.0
+    
+    Parameters
+    ----------
+    score : Score
+        A Score object containing the melody to analyze. The score will be
+        flattened and collapsed into a single sequence of notes ordered by
+        onset time.
+    
+    Returns
+    -------
+    list
+        A list of accent values between 0 and 1 for each note. First and last
+        notes always have accent value 0 as they require three notes for
+        accent calculation.
+    
+    Examples
+    --------
+    >>> score = Score()
+    >>> part = Part()
+    >>> notes = [Note(pitch=60), Note(pitch=64), Note(pitch=62)]  # up-down pattern
+    >>> for note in notes:
+    ...     part.append(note)
+    >>> score.append(part)
+    >>> accent_values = calculate_melodic_accent(score)
+    >>> print(accent_values)
+    [0.0, 1.0, 0.0]  # Peak accent in the middle
+    """
+    # Flatten and collapse the score into a single sequence of notes
+    flattened_score = score.flatten(collapse=True)
+    
+    # Handle empty score
```

Comment:
I don't think we need to deal with this case explicitly, as it is already covered by l. 48-49.